### ISSUE TITLE
pgagroal-cli stop sometimes does not remove processes

### ISSUE THREAD

**Issue by fluca1978** [agroal/pgagroal#198]
I'm using `pgagroal` 1.4.0, after `pgagroal-cli stop` I see there are still processes listed on the machine.

```
$ pgagroal-cli stop
$ ps -auxw | grep pgagroal
pgagroal    2330  0.7  0.3  69936  3624 ?        Ss   04:40   0:18 pgagroal: ma
pgagroal    2484  0.7  0.6  72060  5972 ?        S    04:42   0:17 pgagroal: ma
pgagroal    2485  0.0  0.0      0     0 ?        Z    04:42   0:00 [pgagroal] <defunct>
pgagroal    2577  0.0  0.4  72060  4604 ?        S    04:44   0:00 pgagroal: ma
pgagroal    2578  0.0  0.0      0     0 ?        Z    04:44   0:00 [pgagroal] <defunct>
pgagroal    2579  9.0  0.3  72060  3372 ?        S    04:44   3:32 pgagroal: ma
```

The problem is that if I don't delete (`kill`) those processes, the next time I start `pgagroal` there seems to be some confusion about the instances (e.g., the prefill connections double).
Is this a bug or am I missing something?

**Comment by jesperpedersen** [comment]
Do you have a list of commands that will replicate this issue ?

Can you attach `gdb` to the instances and provide the stack traces ?

**Comment by fluca1978** [comment]
Apparently it was my fault, or I'm not so sure. If you execute twice `pgagroal -d` at any time, the system seems to start another instance that is not going to be stopped:

```
$ pgagroal -d
$ pgagroal -d
$ pgagroal-cli stop
$ pgagroal-cli stop
No connection to pgagroal on /tmp
$ ps - awxf | grep pgagroal
pgagroal    1496  0.0  0.2  69636  2648 ?        Ss   09:45   0:00 pgagroal: ma
$ gdb $(which pgagroal) 1496
...
(gdb) bt
#0  0x00007f57e41079bb in epoll_wait () from /usr/lib64/libc.so.6
#1  0x00007f57e50a77e7 in epoll_poll () from /usr/lib64/libev.so.4
#2  0x00007f57e50a9ca6 in ev_run () from /usr/lib64/libev.so.4
```

**Comment by jesperpedersen** [comment]
Resolved with #201
